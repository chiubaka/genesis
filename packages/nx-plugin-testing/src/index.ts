export * from "./setup";
export * from "./testingWorkspace";
export * from "./verdaccio";
