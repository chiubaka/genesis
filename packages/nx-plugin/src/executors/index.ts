export * from "./nxPluginE2e";
