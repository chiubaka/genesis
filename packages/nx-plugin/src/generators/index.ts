export * from "./linting";
export * from "./preset";
