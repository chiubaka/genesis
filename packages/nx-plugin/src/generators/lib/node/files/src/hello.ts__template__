import crypto from "node:crypto";
import fs from "node:fs";
import os from "node:os";
import path from "node:path";

export const hello = "Hello, world!";

export function writeHelloWorld() {
  const randomSuffix = crypto.randomBytes(48).toString("base64url");
  const fileName = `hello-${randomSuffix}.txt`;

  const tmpDir = os.tmpdir();
  const filePath = path.join(tmpDir, fileName);

  // eslint-disable-next-line security/detect-non-literal-fs-filename
  fs.writeFileSync(filePath, hello);

  return filePath;
}

export class Greeter {
  public hello() {
    return hello;
  }
}
