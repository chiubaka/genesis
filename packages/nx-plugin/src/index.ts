export * from "./generators";
export * from "./utils";
