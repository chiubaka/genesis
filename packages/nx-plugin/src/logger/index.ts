import { Logger, LoggerOptions, LogLevel } from "./Logger";
import { WinstonLogger } from "./winston";

export const createLogger = (
  name: string,
  options: Partial<LoggerOptions> = {},
): Logger => {
  return new WinstonLogger({
    level: LogLevel.Info,
    ...options,
    output: {
      console: true,
      file: [
        {
          path: `${name}.log`,
        },
      ],
      ...options.output,
    },
  });
};

export const generatorLogger = createLogger("generator");
