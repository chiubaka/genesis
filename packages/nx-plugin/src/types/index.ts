export * from "./codecov";
export * from "./dockerCompose";
export * from "./npm";
export * from "./nx";
export * from "./verdaccio";
export * from "./yarn";
