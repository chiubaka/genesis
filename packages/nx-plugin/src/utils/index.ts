export * from "./addDependenciesToPackageJson";
export * from "./exec";
export * from "./getProjectDir";
export * from "./git";
export * from "./github";
export * from "./Project";
export * from "./projectNameCasings";
export * from "./tasks";
