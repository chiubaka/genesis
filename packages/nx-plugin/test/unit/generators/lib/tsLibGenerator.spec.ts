import { createTreeWithEmptyWorkspace } from "@chiubaka/nx-plugin-testing";
import { Tree } from "@nrwl/devkit";

import { Project, tsLibGenerator } from "../../../../src";
import {
  fileMatchesSnapshot,
  nodeProjectTestCases,
  tsProjectTestCases,
} from "../../../cases";

describe("tsLibGenerator", () => {
  let tree: Tree;
  let project: Project;
  let e2eProject: Project;

  let projectName: string;

  const getProject = () => {
    return project;
  };
  const getE2eProject = () => {
    return e2eProject;
  };

  beforeAll(() => {
    tree = createTreeWithEmptyWorkspace();
    project = new Project(tree, "ts-lib", "library");
    e2eProject = new Project(tree, "ts-lib-e2e", "e2e");

    projectName = project.getName();
  });

  tsProjectTestCases(getProject);

  beforeAll(async () => {
    await tsLibGenerator(tree, {
      name: projectName,
      skipE2e: false,
    });
  });

  it("generates a single lib project", () => {
    const projectNames = tree.children(project.relativePath(".."));

    expect(projectNames).toEqual(["ts-lib"]);
  });

  it("generates a single E2E project", () => {
    const e2eProjectNames = tree.children(e2eProject.relativePath(".."));

    // For whatever reason, the Tree directory API doesn't seem 100% correct, so names
    // don't match here (we get `ts-lib-` instead of `ts-lib-e2e`) while they do
    // match in E2E tests.
    expect(e2eProjectNames).toHaveLength(1);
  });

  describe("it generates a code sample", () => {
    it("generates a src/index.ts file", () => {
      // eslint-disable-next-line security/detect-non-literal-fs-filename
      expect(tree.exists(project.srcPath("index.ts"))).toBe(true);
    });

    it("generates a src/hello.ts file", () => {
      // eslint-disable-next-line security/detect-non-literal-fs-filename
      expect(tree.exists(project.srcPath("hello.ts"))).toBe(true);
    });

    it("generates a sample unit test", () => {
      // eslint-disable-next-line security/detect-non-literal-fs-filename
      expect(tree.exists(project.testPath("unit/hello.spec.ts"))).toBe(true);
    });
  });

  describe("E2E project", () => {
    nodeProjectTestCases(getE2eProject, {
      repoName: "ts-lib",
    });

    describe("webpack.config.js", () => {
      it("generates a webpack.config.js file", () => {
        expect(tree.exists(e2eProject.path("webpack.config.js"))).toBe(true);
      });

      fileMatchesSnapshot(
        "webpack.config.js",
        getE2eProject,
        (e2eProject: Project) => {
          return e2eProject.path("webpack.config.js");
        },
      );
    });
  });
});
