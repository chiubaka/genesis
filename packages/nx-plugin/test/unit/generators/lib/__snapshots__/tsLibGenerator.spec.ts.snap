// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tsLibGenerator .eslintrc.json matches snapshot 1`] = `
Object {
  "extends": Array [
    "../../.eslintrc.json",
  ],
  "ignorePatterns": Array [
    "node_modules",
  ],
  "overrides": Array [
    Object {
      "files": Array [
        "jest.config.ts",
        "*.spec.ts",
        "*.test.ts",
      ],
      "parserOptions": Object {
        "project": Array [
          "packages/ts-lib/tsconfig.spec.json",
        ],
      },
    },
    Object {
      "excludedFiles": Array [
        "jest.config.ts",
        "*.spec.ts",
        "*.test.ts",
      ],
      "files": Array [
        "*.ts",
      ],
      "parserOptions": Object {
        "project": Array [
          "packages/ts-lib/tsconfig.lib.json",
        ],
      },
    },
  ],
}
`;

exports[`tsLibGenerator E2E project .eslintrc.json matches snapshot 1`] = `
Object {
  "extends": Array [
    "../../.eslintrc.json",
  ],
  "ignorePatterns": Array [
    "node_modules",
  ],
  "overrides": Array [
    Object {
      "files": Array [
        "jest.config.ts",
        "*.spec.ts",
        "*.test.ts",
      ],
      "parserOptions": Object {
        "project": Array [
          "e2e/ts-lib-e2e/tsconfig.spec.json",
        ],
      },
    },
    Object {
      "excludedFiles": Array [
        "jest.config.ts",
        "*.spec.ts",
        "*.test.ts",
      ],
      "files": Array [
        "*.ts",
      ],
      "parserOptions": Object {
        "project": Array [
          "e2e/ts-lib-e2e/tsconfig.lib.json",
        ],
      },
    },
  ],
}
`;

exports[`tsLibGenerator E2E project jest.config.ts matches snapshot 1`] = `
"export default {
  displayName: \\"ts-lib-e2e\\",
  preset: \\"../../jest.preset.js\\",
    globals: {
    \\"ts-jest\\": {
      tsconfig: \\"<rootDir>/tsconfig.spec.json\\",
    },
  },
      testEnvironment: \\"node\\",
    transform: {
        \\"^.+\\\\\\\\.[tj]s$\\": \\"ts-jest\\",
      },
    moduleFileExtensions: [\\"ts\\", \\"js\\", \\"html\\", \\"json\\"],
    coverageDirectory: \\"../../reports/coverage/packages/ts-lib-e2e\\",
  reporters: [
    \\"default\\",
    [
      \\"jest-junit\\",
      {
        addFileAttribute: \\"true\\",
        classNameTemplate: \\"{suitename}\\",
        outputDirectory: \\"reports/junit\\",
        outputName: \\"ts-lib-e2e.xml\\",
      },
    ],
  ],
};
"
`;

exports[`tsLibGenerator E2E project project.json matches snapshot 1`] = `
Object {
  "$schema": "../../node_modules/nx/schemas/project-schema.json",
  "implicitDependencies": Array [
    "ts-lib",
  ],
  "name": "ts-lib-e2e",
  "projectType": "application",
  "sourceRoot": "e2e/ts-lib-e2e/src",
  "tags": Array [],
  "targets": Object {
    "build": Object {
      "configurations": Object {
        "production": Object {
          "extractLicenses": true,
          "fileReplacements": Array [
            Object {
              "replace": "e2e/ts-lib-e2e/src/environments/environment.ts",
              "with": "e2e/ts-lib-e2e/src/environments/environment.prod.ts",
            },
          ],
          "inspect": false,
          "optimization": true,
        },
      },
      "executor": "@nrwl/webpack:webpack",
      "options": Object {
        "assets": Array [
          "e2e/ts-lib-e2e/src/assets",
        ],
        "compiler": "tsc",
        "main": "e2e/ts-lib-e2e/src/main.ts",
        "outputPath": "dist/e2e/ts-lib-e2e",
        "target": "node",
        "tsConfig": "e2e/ts-lib-e2e/tsconfig.app.json",
      },
      "outputs": Array [
        "{options.outputPath}",
      ],
    },
    "e2e": Object {
      "dependsOn": Array [
        Object {
          "projects": "self",
          "target": "install",
        },
      ],
      "executor": "@nrwl/jest:jest",
      "options": Object {
        "jestConfig": "e2e/ts-lib-e2e/jest.config.ts",
        "passWithNoTests": true,
      },
      "outputs": Array [
        "reports/coverage/e2e/ts-lib-e2e",
        "reports/junit/ts-lib-e2e.xml",
      ],
    },
    "install": Object {
      "dependsOn": Array [
        Object {
          "projects": "dependencies",
          "target": "build",
        },
      ],
      "executor": "nx:run-commands",
      "options": Object {
        "commands": Array [
          "yarn cache clean --all",
          "yarn install",
        ],
        "cwd": "e2e/ts-lib-e2e",
        "parallel": false,
      },
    },
    "lint": Object {
      "executor": "@nrwl/linter:eslint",
      "options": Object {
        "lintFilePatterns": Array [
          "e2e/ts-lib-e2e",
        ],
      },
      "outputs": Array [
        "{options.outputFile}",
      ],
    },
  },
}
`;

exports[`tsLibGenerator E2E project tsconfig.app.json or tsconfig.lib.json matches snapshot 1`] = `
Object {
  "compilerOptions": Object {
    "declaration": true,
    "types": Array [
      "node",
    ],
  },
  "exclude": Array [
    "jest.config.ts",
    "**/*.spec.ts",
    "**/*.test.ts",
    "test",
  ],
  "extends": "./tsconfig.json",
  "include": Array [
    "**/*.ts",
  ],
}
`;

exports[`tsLibGenerator E2E project tsconfig.json matches snapshot 1`] = `
Object {
  "compilerOptions": Object {
    "lib": Array [
      "es2022",
    ],
    "module": "commonjs",
    "outDir": "../../dist/out-tsc",
    "target": "es2022",
  },
  "extends": "../../tsconfig.base.json",
  "files": Array [],
  "include": Array [],
  "references": Array [
    Object {
      "path": "./tsconfig.lib.json",
    },
    Object {
      "path": "./tsconfig.spec.json",
    },
  ],
}
`;

exports[`tsLibGenerator E2E project tsconfig.spec.json matches snapshot 1`] = `
Object {
  "compilerOptions": Object {
    "types": Array [
      "jest",
      "node",
    ],
  },
  "extends": "./tsconfig.json",
  "include": Array [
    "jest.config.ts",
    "**/*.spec.ts",
    "**/*.test.ts",
    "**/*.d.ts",
    "test",
  ],
}
`;

exports[`tsLibGenerator jest.config.ts matches snapshot 1`] = `
"export default {
  displayName: \\"ts-lib\\",
  preset: \\"../../jest.preset.js\\",
    globals: {
    \\"ts-jest\\": {
      tsconfig: \\"<rootDir>/tsconfig.spec.json\\",
    },
  },
      testEnvironment: \\"node\\",
    transform: {
        \\"^.+\\\\\\\\.[tj]s$\\": \\"ts-jest\\",
      },
    moduleFileExtensions: [\\"ts\\", \\"js\\", \\"html\\", \\"json\\"],
    coverageDirectory: \\"../../reports/coverage/packages/ts-lib\\",
  reporters: [
    \\"default\\",
    [
      \\"jest-junit\\",
      {
        addFileAttribute: \\"true\\",
        classNameTemplate: \\"{suitename}\\",
        outputDirectory: \\"reports/junit\\",
        outputName: \\"ts-lib.xml\\",
      },
    ],
  ],
};
"
`;

exports[`tsLibGenerator project.json matches snapshot 1`] = `
Object {
  "$schema": "../../node_modules/nx/schemas/project-schema.json",
  "name": "ts-lib",
  "projectType": "library",
  "sourceRoot": "packages/ts-lib/src",
  "tags": Array [],
  "targets": Object {
    "build": Object {
      "executor": "@nrwl/js:tsc",
      "options": Object {
        "assets": Array [
          "packages/ts-lib/*.md",
        ],
        "main": "packages/ts-lib/src/index.ts",
        "outputPath": "dist/packages/ts-lib",
        "tsConfig": "packages/ts-lib/tsconfig.lib.json",
      },
      "outputs": Array [
        "{options.outputPath}",
      ],
    },
    "deploy": Object {
      "configurations": Object {
        "development": Object {
          "dryRun": true,
        },
        "production": Object {
          "dryRun": false,
        },
      },
      "defaultConfiguration": "development",
      "dependsOn": Array [
        Object {
          "projects": "self",
          "target": "build",
        },
      ],
      "executor": "@chiubaka/nx-plugin:npm-publish",
      "options": Object {
        "packagePath": "dist/packages/ts-lib",
      },
    },
    "lint": Object {
      "executor": "@nrwl/linter:eslint",
      "options": Object {
        "lintFilePatterns": Array [
          "packages/ts-lib",
        ],
      },
      "outputs": Array [
        "{options.outputFile}",
      ],
    },
    "publish": Object {
      "dependsOn": Array [
        Object {
          "projects": "self",
          "target": "build",
        },
      ],
      "executor": "@nrwl/workspace:run-commands",
      "options": Object {
        "command": "node tools/scripts/publish.mjs ts-lib {args.ver} {args.tag}",
      },
    },
    "test": Object {
      "executor": "@nrwl/jest:jest",
      "options": Object {
        "jestConfig": "packages/ts-lib/jest.config.ts",
        "passWithNoTests": true,
      },
      "outputs": Array [
        "reports/coverage/packages/ts-lib",
        "reports/junit/ts-lib.xml",
      ],
    },
  },
}
`;

exports[`tsLibGenerator tsconfig.app.json or tsconfig.lib.json matches snapshot 1`] = `
Object {
  "compilerOptions": Object {
    "declaration": true,
  },
  "exclude": Array [
    "jest.config.ts",
    "**/*.spec.ts",
    "**/*.test.ts",
    "test",
  ],
  "extends": "./tsconfig.json",
  "include": Array [
    "**/*.ts",
  ],
}
`;

exports[`tsLibGenerator tsconfig.json matches snapshot 1`] = `
Object {
  "compilerOptions": Object {
    "lib": Array [
      "es2015",
    ],
    "module": "commonjs",
    "outDir": "../../dist/out-tsc",
    "target": "es2015",
  },
  "extends": "../../tsconfig.base.json",
  "files": Array [],
  "include": Array [],
  "references": Array [
    Object {
      "path": "./tsconfig.lib.json",
    },
    Object {
      "path": "./tsconfig.spec.json",
    },
  ],
}
`;

exports[`tsLibGenerator tsconfig.spec.json matches snapshot 1`] = `
Object {
  "compilerOptions": Object {
    "types": Array [
      "jest",
      "node",
    ],
  },
  "extends": "./tsconfig.json",
  "include": Array [
    "jest.config.ts",
    "**/*.spec.ts",
    "**/*.test.ts",
    "**/*.d.ts",
    "test",
  ],
}
`;
